deepchem/rl/tests/test_mcts.py

```diff
@@ -0,0 +1,89 @@
+from flaky import flaky
+
+import deepchem as dc
+from deepchem.models.tensorgraph.layers import Reshape, Variable, SoftMax, GRU, Dense
+from deepchem.models.tensorgraph.optimizers import Adam, PolynomialDecay
+import numpy as np
+import tensorflow as tf
+import unittest
+from nose.plugins.attrib import attr
+
+
+class TestMCTS(unittest.TestCase):
+
+  @flaky
+  def test_roulette(self):
+    """Test training a policy for the roulette environment."""
+
+    # This is modeled after the Roulette-v0 environment from OpenAI Gym.
+    # The player can bet on any number from 0 to 36, or walk away (which ends the
+    # game).  The average reward for any bet is slightly negative, so the best
+    # strategy is to walk away.
+
+    class RouletteEnvironment(dc.rl.Environment):
+
+      def __init__(self):
+        super(RouletteEnvironment, self).__init__([(1,)], 38)
+        self._state = [np.array([0])]
+
+      def step(self, action):
+        if action == 37:
+          self._terminated = True  # Walk away.
+          return 0.0
+        wheel = np.random.randint(37)
+        if wheel == 0:
+          if action == 0:
+            return 35.0
```

Comment:
Whoever wrote that was mistaken.  Here's the implementation of it: https://github.com/openai/gym/blob/master/gym/envs/toy_text/roulette.py

Reply:
Thanks for the link, though that implementation is super confusing. It say it returns 35, but it actually returns 36. Since in the constructor:
```
def __init__(self, spots=37):
  self.n = spots + 1
```
then in _step:
```
if val == action == 0:
  reward = self.n - 2.0
```